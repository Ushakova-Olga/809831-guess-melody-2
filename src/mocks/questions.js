const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `https://sefan.ru/music/files/eng/low/Leona_Lewis_-_I_See_You_(OST_Avatar).mp3`,
        genre: `rock`,
      },
      {
        src: `https://sefan.ru/music/files/eng/low/Celine_Dion_-_My_Heart_Will_Go_On_(OST_Titanic).mp3`,
        genre: `pop`,
      },
      {
        src: `https://sefan.ru/music/files/rus/low/Nautilus_Pompilius_-_Mater_bogov_(OST_Brat).mp3`,
        genre: `jazz`,
      },
      {
        src: `https://sefan.ru/music/files/eng/low/Pirates_of_the_Caribbean_-_He-s_a_Pirate.mp3`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `genre`,
    genre: `folk`,
    answers: [
      {
        src: `https://sefan.ru/music/files/rus/low/Okudzhava_Bulat_-_Pesnya_Vereschagina_(Vashe_blagorodie_OST_Beloe_solnce_pustyni).mp3`,
        genre: `rock`,
      },
      {
        src: `https://sefan.ru/music/files/eng/low/Avril_Lavigne_-_Alice.mp3`,
        genre: `pop`,
      },
      {
        src: `https://sefan.ru/music/files/eng/low/Pirates_of_the_Caribbean_-_The_Black_Pearl_(Chernaya_Zhemchuzhina).mp3`,
        genre: `jazz`,
      },
      {
        src: `https://sefan.ru/music/files/eng/low/Roy_Orbison_-_Oh_Pretty_Woman.mp3`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Jim Beam`,
      src: `https://sefan.ru/music/files/eng/low/Leona_Lewis_-_I_See_You_(OST_Avatar).mp3`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Jack Daniels`,
      src: `https://sefan.ru/music/files/eng/low/Roy_Orbison_-_Oh_Pretty_Woman.mp3`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
];

export default questions;
