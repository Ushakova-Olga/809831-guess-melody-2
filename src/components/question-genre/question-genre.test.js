import React from "react";
import renderer from 'react-test-renderer';
import QuestionGenre from '../question-genre/question-genre';
import questions from "../../mocks/questions.js";

it(`Question genre screen correctly renders`, () => {

  const tree = renderer
    .create(<QuestionGenre
      screenIndex={1}
      question={questions[0]}
      renderPlayer={() => {}}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
