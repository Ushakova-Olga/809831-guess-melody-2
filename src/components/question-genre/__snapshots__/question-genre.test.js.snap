// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question genre screen correctly renders 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    rock
     треки
  </h2>
  <form
    className="game__tracks"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-rock0"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-rock0"
        />
        <label
          className="game__check"
          htmlFor="answer-rock0"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-rock1"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-rock1"
        />
        <label
          className="game__check"
          htmlFor="answer-rock1"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-rock2"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-rock2"
        />
        <label
          className="game__check"
          htmlFor="answer-rock2"
        >
          Отметить
        </label>
      </div>
    </div>
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          id="answer-rock3"
          name="answer"
          onChange={[Function]}
          type="checkbox"
          value="answer-rock3"
        />
        <label
          className="game__check"
          htmlFor="answer-rock3"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
